zerver/models.py

```diff
@@ -2294,6 +2294,9 @@ class AbstractUserMessage(models.Model):
         # Whether we've sent a push notification to the user's mobile
         # devices for this message that has not been revoked.
         "active_mobile_push_notification",
+        # When a message can be edited by the user that is set as a
+        # publicly editable message, this flag is set
+        "publicly_editable",
```

Comment:
If I understand correctly, `publicly_editable` should be a property of the `Message` object, and not the `UserMessage` object(s).
`UserMessage` is just a link to the main `Message` object, for each user who has received the message, with certain other user-specific data (like mentions). If we want everyone to edit a **message**, I believe the flag to say so should be on the `Message` object itself.

Reply:
(See for example the `has_attachment` or `has_link` flags on `AbstractUserMessage` class)

Reply:
No, I decided to keep it a message flag, so that if we want to set different role-specific permissions or user-specific permissions or say, assign only 2 or 3 people to edit, etc, this structure can help us in the future.

Reply:
@timabbott would you mind throwing some light on what you think is a better implementation? Keeping the future of the issue in mind.

Reply:
Oh! The permissions part makes sense, but I still think this is more of a `Message` thing. For example, if we want to make a message editable by everyone, how would this handle users who haven't subscribed to that stream (and hence don't have a `UserMessage` row for that message)?
We could go with the historical message approach we use to star/react to messages from unsubscribed streams, but I'm not sure if that's the best thing to do.

Reply:
> Oh! The permissions part makes sense, but I still think this is more of a `Message` thing. For example, if we want to make a message editable by everyone, how would this handle users who haven't subscribed to that stream (and hence don't have a `UserMessage` row for that message)?

If the user is not subscribed to a stream then there is almost no way to set user-level permissions for a particular message anyways.

> We could go with the historical message approach we use to star/react to messages from unsubscribed streams, but I'm not sure if that's the best thing to do.

Fundamentally, the star is something that all users set for themselves, we are kind of doing the same but allowing the creator of the message to set it for others.

Reply:
Would definitely say you have a good point, we can actually have 2 flags, one is a `Message` flag for anyone, and the other is a `UserMessage` flag to set user-level permissions.